README.md

```diff
@@ -16,40 +16,135 @@ This code for sample application is intended for demonstration purposes only. It
 * kubectl is installed - https://docs.aws.amazon.com/eks/latest/userguide/install-kubectl.html
 * eksctl is installed - https://docs.aws.amazon.com/eks/latest/userguide/eksctl.html
 * jq is installed - https://jqlang.github.io/jq/download/
+* [Optional] If you plan to install the infrastructure resources using Terraform, terraform cli is required. https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli
 
 # EKS demo
 
-## Build the sample application images and push to ECR
+## Deploy via Shell Scripts
+
+### Build the sample application images and push to ECR
+
 1. Build container images for each micro-service application
 
-```
+``` shell
+
 ./mvnw clean install -P buildDocker
 ```
 
 2. Create an ECR repo for each micro service and push the images to the relevant repos. Replace the aws account id and the AWS Region.
-    
-```
-export ACCOUNT='111122223333'
+
+``` shell
+export ACCOUNT=`aws sts get-caller-identity | jq .Account -r`
 export REGION='us-east-1'
 ./push-ecr.sh
 ```
 
-## Try Application Signals with the sample application
+### Try Application Signals with the sample application
 
-1. Create an EKS cluster, enable Application Signals, and deploy the sample application to your EKS cluster. Replace `new-cluster-name` with the name that you want to use for the new cluster. Replace `region-name` with the same region in previous section "**Build the sample application images and push to ECR**". 
+1. Create an EKS cluster, enable Application Signals, and deploy the sample application to your EKS cluster. Replace `new-cluster-name` with the name that you want to use for the new cluster. Replace `region-name` with the same region in previous section "**Build the sample application images and push to ECR**".
 
-```
+``` shell
 cd scripts/eks/appsignals/one-step && ./setup.sh new-cluster-name region-name
 ```
 
 2. Clean up all the resources. Replace `new-cluster-name` and `region-name` with the same values that you use in previous step.
 
-```
+``` shell
 cd scripts/eks/appsignals/one-step && ./cleanup.sh new-cluster-name region-name
 ```
 
 Please be aware that this sample application includes a publicly accessible Application Load Balancer (ALB), enabling easy interaction with the application. If you perceive this public ALB as a security risk, consider restricting access by employing [security groups](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-update-security-groups.html).
 
+## Deploy via Terraform
+
+1. Go to the terraform directory under the project. Prepare Terraform S3 backend and set required environment variables
+
+   ``` shell
+   cd terraform/eks
+
+   # create a new bucket for terraform backend storage
+   aws s3 mb s3://tfstate-$(uuidgen)
```

Comment:
On OSX, this will fail since it uses uppercase letters (only linux uses lowercase). Since people may try this locally let's convert it. `aws s3 mb s3://tfstate-$(uuidgen | tr A-Z a-z)`.